app/util/data_preparation/confluence_prepare_data.py

```diff
@@ -21,17 +24,44 @@
 ENGLISH_GB = 'en_GB'
 
 
+def print_timing(message, sep='-'):
+    assert message is not None, "Message is not passed to print_timing decorator"
+
+    def deco_wrapper(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs) -> object:
+            start = timer()
+            print(sep * 20)
+            print(f'{message} started {datetime.datetime.now().strftime("%H:%M:%S")}')
+            result = func(*args, **kwargs)
+            end = timer()
+            print(f"{message} finished in {timedelta(seconds=end - start)}")
+            print(sep * 20)
+            return result
+
+        return wrapper
+
+    return deco_wrapper
+
+
 def generate_random_string(length=20):
     return "".join([random.choice(string.ascii_lowercase) for _ in range(length)])
 
 
+@print_timing('Creating dataset started')
 def __create_data_set(rest_client, rpc_client):
     dataset = dict()
     dataset[USERS] = __get_users(rest_client, rpc_client, CONFLUENCE_SETTINGS.concurrency)
     perf_user = random.choice(dataset[USERS])['user']
     perf_user_api = ConfluenceRestClient(CONFLUENCE_SETTINGS.server_url, perf_user['username'], DEFAULT_USER_PASSWORD)
-    dataset[PAGES] = __get_pages(perf_user_api, 5000)
-    dataset[BLOGS] = __get_blogs(perf_user_api, 5000)
+
+    def __get_pages_blogs(func):
+        return func(perf_user_api, 5000)
```

Comment:
With this approach, we break the possibility to modify the number of pages and blogs separately.
Also, the code is quite hard to understand.

Let's try something more simple and straight forward like:
```
    pool = ThreadPool(processes=2)
    async_pages = pool.apply_async(__get_pages, (perf_user_api, 5000))
    async_blogs = pool.apply_async(__get_blogs, (perf_user_api, 5000))

    async_pages.wait()
    async_blogs.wait()

    dataset[PAGES] = async_pages.get()
    dataset[BLOGS] = async_blogs.get()
```